text/0000-light-dom.md

```diff
@@ -0,0 +1,330 @@
+
+## Detailed design
+
+### Selecting Light DOM vs Shadow DOM
+
+The selection of Light DOM vs Shadow DOM is under the control of the component developer. It is done at the component class level, by inheriting a different base class (`MacroElement`, not `LightningElement`). The LWC compiler can also use this information to check any issue and warn the developer.
+
+```js
+import { MacroElement } from "lwc";
+
+export default class MyComponent extends MacroElement {}
+```
+
+### Component features when using Light DOM
+
+Some of the LWC component capabilities are directly inherited from Shadow DOM, or emulated by the synthetic-shadow. Despite the use of Light DOM, we’d like to keep these features available, even if their behavior is slightly adapted to the Light DOM:
+
+- **Slots**
+
+  As we mentioned before, the component composition model in LWC is provided by slots. Light DOM will provide the same mental model for developers building Light DOM components.
+
+  In Light DOM, `<slot>` will denote the place where the slotted component will be attached. The `<slot>` element itself won't be rendered. The slotted content (or the placeholder content) will be flattened to the parent element at runtime.
+
+  Since the `<slot>` element itself isn't rendered, adding any event listeners to the `<slot>` element in the template will throw a compiler error.
+
+- **Scoped Styles**
+
+  Shadow DOM styles are scoped to the component they belong. In Native shadow, it is enforced by the browser while in Synthetic shadow the styles are scoped by some attributes which are also added to the element at runtime.
+
+  In Light DOM, the styles will be scoped not just to the component they belong, but also to its children. This is similar to how Aura styles its components.
+
+  E.g.
+
+  ```javascript
+  <x-a>
+    <style>
+      p { color: blue }
+    </style>
+
+    <p item=1></p>
+
+    <x-b>
+      <style>
+        p { color: red }
+      </style>
+
+      <p item=2></p>
+      <p item=3></p> <!-- This paragraph is coming from A and slotted into B -->
+    </x-b>
+  </x-a>
+  ```
+
+  In this case, item 1 will be in blue and item 2 and 3 will be in red. Also note that in `<x-a>`, we can write a selector of higher specificity `x-b p` to override the styles in `x-b`.
+
+  Global scoping is not supported. If desired, a global stylesheet can be injected manually.
+
+- **`this.template`**
+
+  In `LightningElement`, `this.template` returns the shadow-root. It will return `null` in `MacroElement`.
+
+### Security (WIP)
+
+- In some applications, light-dom components may not be allowed... it’s up to the app context
+  - **what is the behavior when it’s not allowed?**
+  - Some applications might disable light-dom as a whole
+  - Some applications might disable light-dom selectively using a “privileged code” model
+
+### Component Migration
+
+There is no migration of the existing components needed. The behaviors of existing components using Shadow DOM remain the same.
```

Comment:
I would tie this into the new `MacroElement` model saying components extending from `LightningElement` have nothing to do 9of that is the case)